gcloud/datastore/connection.py

```diff
@@ -312,6 +312,7 @@ def commit(self, dataset_id, mutation_pb, transaction_id):
                    that was completed in the commit.
         """
         request = _datastore_pb2.CommitRequest()
+        request.CopyFrom(commit_request)
```

Comment:
To think about re-use or to think about the fact that we never thought about re-use? We do [tombstone][1] a `Transaction` and have [tests][2] to ensure this stays true.

[1]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/ee25a8d4330557e6f64867c5d22ab543a90780a9/gcloud/datastore/transaction.py#L100-L101
[2]: https://github.com/GoogleCloudPlatform/gcloud-python/blob/ee25a8d4330557e6f64867c5d22ab543a90780a9/gcloud/datastore/test_transaction.py#L76